lib/dynamoid/transaction/delete.rb

```diff
@@ -0,0 +1,35 @@
+require_relative 'action'
+
+module Dynamoid
+  class Transaction
+    class Delete < Action
+
+      def initialize(model_or_model_class, id = nil, options = {})
+        model = model_or_model_class.is_a?(Dynamoid::Document) ?
+                  model_or_model_class : model_or_model_class.find(id, consistent_read: true)
+        super(model, attributes, options)
+      end
+
+      def run_callbacks
+        model.run_callbacks(:destroy) do
+          yield if block_given?
+        end
+      end
+
+      def to_h
+        key = {model_class.hash_key => model.hash_key}
+        key[model_class.range_key] = model.range_key if model_class.range_key?
+        # force fast fail i.e. won't retry if record is missing
+        condition_expression = "attribute_exists(#{model_class.hash_key})"
+        condition_expression += " and attribute_exists(#{model_class.range_key})" if model_class.range_key? # needed?
```

Comment:
Not sure we need to check whether item exists. If the condition fails the whole transaction will fails. I would like to have behaviour similar to Rails as mach as possible.

Reply:
I'm adding support for delete, destroy and destroy! The destroys will keep the existence check, delete will not.